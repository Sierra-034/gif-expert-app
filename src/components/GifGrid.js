import React from "react";

export const GifGrid = ({ category }) => {

    const getGifs = async () => {
        const URL = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&limit=10&api_key=4N0WEqRJUE4MR2NNALGC8qIYVvuNUXwM';
        const response = await fetch(URL);
        const {data} = await response.json();

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        
        console.log(gifs);
    };

    getGifs();
    
    return (
        <div>
            <h3>{ category }</h3>
        </div>
    );
};