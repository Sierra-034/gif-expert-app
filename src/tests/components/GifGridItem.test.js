import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas sobre <GifGridItem />', () => {

    const title = 'Un título';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    
    test('Muestra el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Tener un párrafo con el título', () => {
        const p = wrapper.find('p');
        expect(p.text()).toBe(title);
    });

    test('Tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('Tiene la clase css animate__fadeIn', () => {
        const expectedCssClass = 'animate__fadeIn';
        const div = wrapper.find('div');
        expect(div.hasClass(expectedCssClass)).toBe(true);
    });
});