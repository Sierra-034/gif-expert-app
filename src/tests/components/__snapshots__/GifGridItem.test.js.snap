// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifGridItem /> Muestra el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-1s"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
